relay-server/src/endpoints/common.rs

```diff
@@ -475,20 +476,24 @@ where
                         .map(|project| (envelope, rate_limits, Some(project)));
                     Box::new(response) as RetVal
                 }))
-                .and_then(|(envelope, rate_limits, sampling_project)| {
+                .and_then(move |(envelope, rate_limits, sampling_project)| {
                     // do the fast path transaction sampling (if we can't do it here
                     // we'll try again after the envelope is queued)
                     let event_id = envelope.event_id();
 
-                    utils::sample_transaction(envelope, sampling_project.clone(), true).then(
-                        move |result| match result {
-                            Err(()) => Err(BadStoreRequest::TraceSampled(event_id)),
-                            Ok(envelope) if envelope.is_empty() => {
-                                Err(BadStoreRequest::TraceSampled(event_id))
-                            }
-                            Ok(envelope) => Ok((envelope, rate_limits, sampling_project)),
-                        },
+                    utils::sample_transaction(
+                        envelope,
+                        sampling_project.clone(),
+                        true,
+                        processing_enabled,
                     )
+                    .then(move |result| match result {
+                        Err(()) => Err(BadStoreRequest::TraceSampled(event_id)),
```

Comment:
Since this is not a client error, we should map this "error" to a _2XX_ status code. I would propose _202 Accepted_.

Reply:
At the moment it is caught in the or_else branch (at line 518) and then it is converted to a 200 by the `create_response` closure. If we really want a 202 (which will only be returned on the 'fast path') we need
to change  `handle_store_like_request`, do we really want to do that ?

Reply:
It should be enough to just add a new match arm in the [`ResponseError` implementation](https://github.com/getsentry/relay/blob/340673f1da25f0f342159e33c7381393e8c7a59a/relay-server/src/endpoints/common.rs#L144).